sky/serve/serve_utils.py

```diff
@@ -1581,7 +1696,19 @@ def _format_replica_table(replica_records: List[Dict[str, Any]], show_all: bool,
         replica_status = record['status']
         status_str = replica_status.colored_str()
         used_by = record.get('used_by', None)
-        used_by_str = str(used_by) if used_by is not None else '-'
+        if used_by is None:
+            used_by_str = '-'
+        else:
+            if len(used_by) > 2:
+                used_by_str = (
+                    f'{used_by[0]}, {used_by[1]}, +{len(used_by) - 2}'
+                    ' more')
+            elif len(used_by) == 2:
+                used_by_str = f'{used_by[0]}, {used_by[1]}'
+            elif len(used_by) == 1:
+                used_by_str = str(used_by[0])
+            else:
+                used_by_str = '-'
```

Comment:
We also need to support the case where used_by is a str for backwards compatibility I think.